core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -195,6 +195,16 @@ default ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connec
         return emptyList();
     }
 
+    /**
+     * Gets all of the columns on the specified table, or an empty list if the columns cannot be enumerated.
+     *
+     * @throws RuntimeException if table handle is no longer valid
+     */
+    default List<ColumnHandle> getColumns(ConnectorSession session, ConnectorTableHandle tableHandle)
```

Comment:
Per commit message, this is complementary to `Map<String, ColumnHandle> getColumnHandles` -- it reduces less information, but provides an explicit ordering.

Do we want to maintain the two methods for ever?
Or can we consider deprecating one of them and removing over time?
The design proposed here does not allow this (none of the variants is strictly superior over the other), but maybe there are other options we could consider?

Reply:
> Do we want to maintain the two methods for ever?

May be yes, but I have an idea to introduce another method 
```
Map<ColumnHandle, ColumnMetadata> getColumnMetadatas
``` 
which probably can be used further to emulate 
```
Map<String, ColumnHandle> getColumnHandles
```
with path `ColumnHandle` -> `ColumnMetadata` -> `column_name` -> Map<column_name, ColumnHandle>
but it is to investigate if it brings performance issues.